blivet/udev.py

```diff
@@ -592,15 +597,12 @@ def device_get_lv_type(info):
 
 def device_dm_subsystem_match(info, subsystem):
     """ Return True if the device matches a given device-mapper subsystem. """
-    uuid = info.get("DM_UUID", "")
-    uuid_fields = uuid.split("-")
-    _subsystem = uuid_fields[0]
-    if _subsystem.lower().startswith("part") and len(uuid_fields) > 1:
-        # kpartx uses partN- as a subsystem prefix, which we ignore because
-        # we only care about the subsystem of the partitions' parent device.
-        _subsystem = uuid_fields[1]
-
-    if _subsystem == uuid or not _subsystem:
+    name = info.get("DM_NAME")
+    if name is None:
+        return False
+
+    _subsystem = blockdev.dm_get_subsystem_from_name(name)
```

Comment:
This should be ``blockdev.dm.get_subtsystem_from_name()``. ``blockdev.dm`` is a so-called ``ErrorProxy`` making sure nice python exceptions are returned instead of the raw C ones.

Reply:
Fixed.